geometry/area_of_polygon.cpp

```diff
@@ -0,0 +1,33 @@
+/*
+The area of a polygon
+Input:
+  The first line contains one number N (3 <= N <= 100 000). The reference is to be reliedly
+  bypass clockwise or counterclockwise.
+  All coordinates are integers, modulo not exceeding 10^4.
+Output:
+  One number is the area of the reduced polygon.
+*/
+
+#include <bits/stdc++.h>
+using namespace std;
+int main()
```

Comment:
Separate the code into functions and structures if needed as well as add self-test cases using `assert` (requires the `cassert` library/header).